src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -150,66 +149,62 @@ private Collector getHybridQueryCollector() {
+        final List<HybridSearchCollector> hybridSearchCollectors = getHybridSearchCollectors(collectors);
+        if (hybridSearchCollectors.isEmpty()) {
+            throw new IllegalStateException("cannot collect results of hybrid search query, there are no proper score collectors");
         }
+        return reduceSearchResults(getSearchResults(hybridSearchCollectors));
+    }
 
-        if (!hybridTopScoreDocCollectors.isEmpty()) {
-            HybridTopScoreDocCollector hybridTopScoreDocCollector = hybridTopScoreDocCollectors.stream()
-                .findFirst()
-                .orElseThrow(() -> new IllegalStateException("cannot collect results of hybrid search query"));
-            List<TopDocs> topDocs = hybridTopScoreDocCollector.topDocs();
-            TopDocs newTopDocs = getNewTopDocs(
-                getTotalHits(this.trackTotalHitsUpTo, topDocs, isSingleShard, hybridTopScoreDocCollector.getTotalHits()),
-                topDocs
-            );
-            TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(newTopDocs, hybridTopScoreDocCollector.getMaxScore());
-            return (QuerySearchResult result) -> { result.topDocs(topDocsAndMaxScore, getSortValueFormats(sortAndFormats)); };
+    private List<ReduceableSearchResult> getSearchResults(final List<HybridSearchCollector> hybridSearchCollectors) {
+        List<ReduceableSearchResult> results = new ArrayList<>();
+        DocValueFormat[] docValueFormats = getSortValueFormats(sortAndFormats);
+        for (HybridSearchCollector collector : hybridSearchCollectors) {
+            TopDocsAndMaxScore topDocsAndMaxScore = getTopDocsAndAndMaxScore(collector, docValueFormats);
+            results.add((QuerySearchResult result) -> reduceCollectorResults(result, topDocsAndMaxScore, docValueFormats));
         }
+        return results;
+    }
 
-        // TODO: Cater the fix for the Bug https://github.com/opensearch-project/neural-search/issues/799
-        if (!hybridSortedTopDocCollectors.isEmpty()) {
-            HybridTopFieldDocSortCollector hybridSortedTopScoreDocCollector = hybridSortedTopDocCollectors.stream()
-                .findFirst()
-                .orElseThrow(() -> new IllegalStateException("cannot collect results of hybrid search query"));
-
-            List<TopFieldDocs> topFieldDocs = hybridSortedTopScoreDocCollector.topDocs();
-            long maxTotalHits = hybridSortedTopScoreDocCollector.getTotalHits();
-            float maxScore = hybridSortedTopScoreDocCollector.getMaxScore();
-
-            TopDocs newTopDocs = getNewTopFieldDocs(
-                getTotalHits(this.trackTotalHitsUpTo, topFieldDocs, isSingleShard, maxTotalHits),
-                topFieldDocs,
+    private TopDocsAndMaxScore getTopDocsAndAndMaxScore(
+        final HybridSearchCollector hybridSearchCollector,
+        final DocValueFormat[] docValueFormats
+    ) {
+        TopDocs newTopDocs;
+        List<? extends TopDocs> topDocs = hybridSearchCollector.topDocs();
```

Comment:
Why we cannot use raw here? It would be simpler, `? extends TopdDoc` is already part of the `topDocs` method definition. This way you can also avoid cast to List<TopDocs> at 184

Reply:
No we can use raw here because the object return by hybridSearchCollector.topDocs is a list of an object. The getTopFieldDocs and getTopDocs method expect arguments in the specific form not in generic form. Therefore, I cannot use raw here.

Reply:
below syntax should work with collection of raw objects:
```
List topDocs = hybridSearchCollector.topDocs();
        if (docValueFormats != null) {
            newTopDocs = getNewTopFieldDocs(
                getTotalHits(this.trackTotalHitsUpTo, topDocs, hybridSearchCollector.getTotalHits()),
                topDocs,
                sortAndFormats.sort.getSort()
            );
        } else {
            newTopDocs = getNewTopDocs(getTotalHits(this.trackTotalHitsUpTo, topDocs, hybridSearchCollector.getTotalHits()), topDocs);
        }
```